plugin.rb

```diff
@@ -450,29 +458,27 @@ def accepted_answer
     end
 
     def accepted_answer_post_info
-      # TODO: we may already have it in the stream ... so bypass query here
-      postInfo =
-        Post
-          .where(id: accepted_answer_post_id, topic_id: object.topic.id)
-          .joins(:user)
-          .pluck("post_number", "username", "cooked", "name")
-          .first
-
-      if postInfo
-        postInfo[2] = if SiteSetting.solved_quote_length > 0
-          PrettyText.excerpt(postInfo[2], SiteSetting.solved_quote_length, keep_emoji_images: true)
+      post_info =
+        if post = object.posts.find { |p| p.post_number == accepted_answer_post_id }
```

Comment:
This is the "performance" change which looks up the stream for the post that is marked as the solution before doing a request to the database.